### PR TITLE
Set sensible defaults for pkg_pconfig_dirs plan variable

### PR DESCRIPTION
**Pull request by reset**
Prior to these changes the default value for pkg_pconfig_dirs was an
empty array of strings. A plan author could override this variable
with directories containing `.pc` files which output the metafile
`PKG_CONFIG_PATH`. This path is propagated throughout the build
process making linking software easier and less error prone in Habitat.
This is useful, but we can do better.

We will now provide a better default if the plan author does not set
`pkg_pconfig_dirs` by attempting to locate a directory at
`$pkg_prefix/lib/pkgconfig` or `$pkg_prefix/share/pkgconfig` and, if
present, set the `pkg_pconfig_dirs` value to one or both of those
locations.

While `lib/pkgconfig` and `share/pkgconfig` are not documented as the
conventional locations by the pkgconfig documentation, it is referred to
as the conventional location by a number of hits from a google search.

This will make linking to software packaged by Habitat easier even if
the plan author didn't explicitly think of documenting pkgconfig dirs.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by stevendanna** [comment]
👍 Really good improvement.  This will help a lot.

File: components/plan-build/bin/hab-plan-build.sh [hunk 1]
```diff
@@ -1883,13 +1885,29 @@ _build_metadata() {
...
-    else
-      pconfig_path_part="${pconfig_path_part}:${pkg_prefix}/${inc}"
-    fi
-  done
+  if [ ${#pkg_pconfig_dirs[@]} -eq 0 ]; then
+    # Plan doesn't define pkg-config paths so let's try to find them in the conventional locations
+    local locations=(lib/pkgconfig share/pkgconfig)
+    for dir in "${locations[@]}"; do
+      if [[ -d "${pkg_prefix}/${dir}" ]]; then
+        echo "WE HAVE ONE OF THESE DIRECTORIES"
```
Comment (smurawski) [inline components/plan-build/bin/hab-plan-build.sh]: Was this just an extra logging statement?

Comment (reset) [inline components/plan-build/bin/hab-plan-build.sh]: Totally, oops!

**Comment by reset** [comment]
@smurawski removed that extra debug output, thanks for the catch!

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/165028246).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/165028246).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by lilianmoraru** [comment]
I guess this makes https://github.com/habitat-sh/core-plans/pull/185 mostly obsolete(PKG_CONFIG_PATH still needs to be removed).

**Comment by reset** [comment]
@lilianmoraru I don't think it makes it absolete at all, if anything this change makes that PR easier to take in (which we will)! Prior, we weren't automatically detecting the pkgconfig dirs so they needed to be set manually and nobody had taken the time yet on the team to verify we had exported the pkgconfig dirs properly for every core package

**Pull request merged**